client/client.go

```diff
@@ -225,6 +220,12 @@ func main() {
 	kingpin.Parse()
 	logger := promlog.New(&promlogConfig)
 	coordinator := Coordinator{logger: logger}
+
+	if myFqdn == nil || strings.TrimSpace(*myFqdn) == "" {
```

Comment:
Don't trim spaces, let's not 2nd guess the user. I don't think this can be nil either.

Reply:
Ok